packages/core/src/browser/frontend-application.ts

```diff
@@ -251,7 +259,7 @@ export class FrontendApplication {
     /**
      * `beforeunload` listener implementation
      */
-    protected preventStop(): boolean {
+    preventStop(): boolean {
```

Comment:
@akosyakov, are you OK exposing this as a public method from the frontend application, or shall I introduce a service for this? Thanks!

Reply:
I would prefer it we abstract `window.addEventListener('beforeunload'` in a new service and implement it for an electron.

Reply:
Maybe `WindowService` can have an event?

Reply:
> I would prefer if we abstract `window.addEventListener('beforeunload'` in a new service

👍 

> and implement it for an electron.

The electron implementation is completely different. From `electron.BrowserWindow.on('close')`:
> Emitted when the window is going to be closed. It's emitted before the beforeunload and unload event of the DOM.

So it means, even if we add any logic for both electron and the browser, the electron `app` will quit because it has already received the close event...

But I got the message and I agree. Let's try not to expose it from the `FrontendApplication`. Thanks!

Reply:
I have made the suggested API changes. From now on, the `WindowService` will now if the window `canUnload`.